packages/bootstrap-vue-next/src/components/BForm/BFormDatalist.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <datalist :id="computedId">
+    <slot name="first" />
+    <template v-for="(option, index) in normalizedOptsWrapper" :key="index">
+      <BFormSelectOption :value="option.value" :disabled="option.disabled">
+        <!-- eslint-disable-next-line vue/no-v-html -->
+        <span v-if="!!option.html" v-html="option.html" />
+        <template v-else>
+          {{ option.text }}
+        </template>
+      </BFormSelectOption>
+    </template>
+    <slot />
+  </datalist>
+</template>
+
+<script setup lang="ts" generic="T">
+import type {BFormDatalistProps, SelectOption} from '../../types'
+import {computed} from 'vue'
+import BFormSelectOption from '../BFormSelect/BFormSelectOption.vue'
+import {useDefaults, useFormSelect, useId} from '../../composables'
+
+const _props = withDefaults(defineProps<BFormDatalistProps>(), {
+  disabled: false,
+  disabledField: 'disabled',
+  htmlField: 'html',
+  id: undefined,
+  // eslint-disable-next-line vue/require-valid-default-prop
+  options: () => [],
+  textField: 'text',
+  valueField: 'value',
+})
+const props = useDefaults(_props, 'BFormDatalist')
+
+defineSlots<{
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  default?: (props: Record<string, never>) => any
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  first?: (props: Record<string, never>) => any
+}>()
+
+const computedId = useId(() => props.id, 'datalist')
+
+const {normalizedOptions, isComplex} = useFormSelect(
+  () => props.options,
+  computed(() => ({...props, optionsField: 'options', labelField: 'label'})).value
```

Comment:
This is wrong. You should be passing in a Ref, not its .value =>

ie. if props changes, useFormSelect wont update

```suggestion
  computed(() => ({...props, optionsField: 'options', labelField: 'label'}))
```